src/Resources/public/gtm.enhancedEcommerce.checkout.js

```diff
@@ -1,19 +1,28 @@
+'use strict';
 (function ($) {
-  'use strict';
 
-  if (typeof checkoutStepsConfiguration !== "undefined" && checkoutStepsConfiguration.enabled === true) {
-    for (var stepId in checkoutStepsConfiguration.steps) {
-      if (checkoutStepsConfiguration.steps.hasOwnProperty(stepId)) {
-        checkoutStepsConfiguration.steps[stepId].forEach(function (conf) {
-          $(conf.selector).on(conf.event, function (e) {
-            var option = null;
-            if (typeof conf.option !== "undefined" && typeof window[conf.option] === 'function') {
-              option = window[conf.option].call(this);
-            }
-            enhancedEcommerceTrackCheckoutOption(stepId, option);
-          });
+  if (typeof checkoutStepsConfiguration === "undefined") return;
+  if (typeof checkoutStepsConfiguration !== "object") return;
+  if (!checkoutStepsConfiguration.hasOwnProperty('enabled')) return;
+  if (checkoutStepsConfiguration.enabled === false) return;
+  if (!checkoutStepsConfiguration.hasOwnProperty('steps')) return;
+
+  for (var stepId in checkoutStepsConfiguration.steps) {
+    if (checkoutStepsConfiguration.steps.hasOwnProperty(stepId)) {
```

Comment:
I would still turn this around:

```
if (!checkoutStepsConfiguration.steps.hasOwnProperty(stepId)) {
    continue;
}

... continue here with other code
```

Makes it easier to read by early returning from the loop in case the conditions don't match. Makes it easier to comprehend, less indentation needed also.